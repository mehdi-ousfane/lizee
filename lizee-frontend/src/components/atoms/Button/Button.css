.button {
  background: blue;
  color: white;
  border: none;
  border-radius: 1rem;
  padding: 0.5rem 1rem;
  font-size: 1.5rem;
  font-weight: bold;
  cursor: pointer;
}

.button:disabled {
  background: hsl(0, 0%, 75%);
  color: grey;
}
