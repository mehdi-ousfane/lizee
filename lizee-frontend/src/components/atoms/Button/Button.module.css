.button {
  background: blue;
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  border: none;
  border-radius: 1rem;
  padding: 0.5rem 1rem;
  cursor: pointer;
  user-select: none;
  overflow: hidden;
  transition-duration: 0.4s;
}

.button:disabled {
  background: hsl(0, 0%, 75%);
  color: grey;
  pointer-events: none;
}

.button:hover:not(.button:disabled) {
  transition-duration: 0.1s;
  background: navy;
}

.button:after {
  content: '';
  background: #ea5381;
  display: block;
  position: absolute;
  border-radius: 1rem;
  padding-top: 160%;
  padding-left: 160%;
  margin-left: -20px !important;
  margin-top: -50%;
  opacity: 0;
  transition: all 0.8s;
}

.button:active:after {
  padding: 0;
  margin: 0;
  opacity: 1;
  transition: 0s;
}
