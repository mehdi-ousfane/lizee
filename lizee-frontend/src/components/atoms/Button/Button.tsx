import { ReactNode, MouseEvent } from 'react';
import styles from './Button.module.css';

interface Props {
  children: ReactNode;
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
  disabled?: boolean;
  addingClass?: string;
}

function Button({ children, onClick, disabled, addingClass = '' }: Props) {
  const classList = [styles.button, addingClass];
  return (
    <button
      className={classList.join(' ')}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
}

export default Button;
