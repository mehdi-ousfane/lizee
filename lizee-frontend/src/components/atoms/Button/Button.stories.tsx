import { Meta } from '@storybook/react/types-6-0';
import Button from './Button';

export default {
  title: 'atoms/Button',
  component: Button,
} as Meta;

const containerStyle = {
  display: 'flex',
  height: '50px',
  width: '80px',
  justifyContent: 'center',
};

export const Active = () => (
  <div style={containerStyle}>
    <Button>Mon bouton</Button>
  </div>
);
export const Inactive = () => (
  <div style={containerStyle}>
    <Button disabled>Mon bouton</Button>
  </div>
);
