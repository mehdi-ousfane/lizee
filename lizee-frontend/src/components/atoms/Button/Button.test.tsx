import { render, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('<Button />', () => {
  test('should match snapshots', () => {
    const { container } = render(<Button>My button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshots when disabled', () => {
    const { container } = render(<Button disabled>My button</Button>);
    expect(container).toMatchSnapshot();
  });

  test('should call onClick prop', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<Button onClick={onClick}>My button</Button>);

    fireEvent.click(getByRole('button'));

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("shouldn't call onClick prop when disabled", () => {
    const onClick = jest.fn();
    const { getByRole } = render(
      <Button onClick={onClick} disabled={true}>
        My button
      </Button>,
    );

    fireEvent.click(getByRole('button'));

    expect(onClick).not.toHaveBeenCalled();
  });
});
