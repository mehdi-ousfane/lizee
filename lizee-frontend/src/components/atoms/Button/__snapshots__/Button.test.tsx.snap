// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshots 1`] = `
<div>
  <button
    class="button "
  >
    My button
  </button>
</div>
`;

exports[`<Button /> should match snapshots when disabled 1`] = `
<div>
  <button
    class="button "
    disabled=""
  >
    My button
  </button>
</div>
`;
